### DISCUSSION TITLE
How to know exactly which configs are/are not overridable?

### DISCUSSION THREAD

**Discussion by jcosentino11** [emqx/emqx#10891] [Q&A] [answered]
From https://www.emqx.io/docs/en/v5.0/admin/cfg.html:
> Some of the configs (such as node.name) are boot-only configs and not overridable. Config values from *-override.conf are not mapped to boot configs for the config fields attributed with mapping: path.to.boot.config.key

Is there an exhaustive list of which 5.x configs are not overridable (I realize this may be a moving target), and/or is there a way for a custom plugin to figure this out programmatically?

**Comment by thalesmg** [comment, marked as answer]
Hi @jcosentino11 .

I'm afraid we don't have a list of only those configurations at the moment.  Those tend to be marked as `readOnly` (e.g.: [here](https://github.com/emqx/emqx/blob/aa98740280e06aed6a3b300866faef961d7eb673/apps/emqx_conf/src/emqx_conf_schema.erl#L141)) and/or map to `vm_args` (e.g.: [here](https://github.com/emqx/emqx/blob/aa98740280e06aed6a3b300866faef961d7eb673/apps/emqx_conf/src/emqx_conf_schema.erl#LL429C10-L429C10)).

I'm not sure there's an easy way to inspect the schema for that.  Perhaps one way to start exploring programmatically exploring the schema would be something along the lines of:

```elixir
{root_ns, root_fields, structs} = :hocon_schema.find_structs(:emqx_conf_schema)
for {_ns, name, meta} <- structs,
    {field_name, %{readOnly: true}} <- meta.fields do
  {name, field_name}
end
```

Note: use with care, as this is something that might change between versions.  This is just a starting, quick and dirty example.
